File: Week 2 - Arrays/pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>
#include <ctype.h>

// init functions
int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    // Get user input
    string text = get_string("Text: ");

    float letters = count_letters(text);
    float words = count_words(text);
    float sentences = count_sentences(text);
    
    float l = (letters / words) * 100;
    float s = (sentences / words) * 100;
    int cl_index = round(0.0588 * l - 0.296 * s - 15.8);

    if (cl_index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (cl_index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", cl_index);
    }
}

int count_letters(string text)
{
    int letters = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        // Count only letter, no numbers oder other chars
        if (isalpha(text[i]))
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string text)
{
    int words = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isspace(text[i]))
        {
            words++;
        }
    }
    // Check, if there is a single word, or nothing
    if (strlen(text) != 0)
    {
        words++;
    }
    return words++;
}

int count_sentences(string text)
{
    int sentences = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentences++;
        }
    }
    // Check, if there is only one sentence without punctuation
    if (sentences == 0 && strlen(text) != 0)
    {
        sentences++;
    }
    return sentences;
}

File: Week 6 - Python/Readability/readability.py
import re


def get_str(text):
    while True:
        try:
            return input(text)
        except:
            print("Please enter text")


# Count letters, and only lettersa
def count_letters(txt):
    count = 0
    for letter in txt:
        if letter.isalpha():
            count += 1
    return count


# Count all words
def count_words(txt):
    return len(txt.split())


# Count sentences via re.split / txt.count(". ") ...
def count_sentences(txt):
    return len(re.split(r'[.!?]+', txt)) - 1


def main():
    while True:
        txt = get_str("Text: ")
        if len(txt) > 0:
            break
        else:
            print("Please type in some text")
            continue

    letters = count_letters(txt)
    words = count_words(txt)
    sentences = count_sentences(txt)

    l = (letters / words) * 100
    s = (sentences / words) * 100
    cl = round(0.0588 * l - 0.296 * s - 15.8)

    if cl < 1:
        print("Before Grade 1")
    elif cl > 16:
        print("Grade 16+")
    else:
        print(f"Grade {cl}")


if __name__ == "__main__":
    main()
